main.py

```diff
@@ -20,9 +20,10 @@
     "apikey": STOCK_KEY
 }
 stock_api_endpoint ="https://www.alphavantage.co/query"
-today = date.today()
 
+# TODO Remove this part it is not needed
 # Check if it is Sunday of Monday
+today = date.today()
 if today.weekday() == 0:
```

Comment:
If day is Tuesday  == 1 yesterday is today -1 but before yesterday would be today - 4 cause on sunday there is no stock data and we need to use friday. But after refacoring it should not be a problem